server/server.go

```diff
@@ -70,9 +70,11 @@ func WithMetrics(handleFn func(http.ResponseWriter, *http.Request) error, m metr
 		// we want to group all requests by commitment header, otherwise the prometheus metric labels will explode
 		commitmentHeader := r.URL.Path[:3]
 		recordDur := m.RecordRPCServerRequest(commitmentHeader)
-		defer recordDur()
 
-		return handleFn(w, r)
+		err := handleFn(w, r)
```

Comment:
hmm just realized something - the `commitmentHeader` will change based on the caller client. (e.g, nitro uses a generic commitment representation which only prepends a version byte).

Reply:
ehh you're saying nitro doesn't follow the 3 byte convention outlined in https://github.com/Layr-Labs/eigenda-proxy?tab=readme-ov-file#commitment-schemas?

Reply:
🤦🏻‍♂️

Reply:
will put up a PR shortly with readme updates - but no it doesn't

Reply:
https://github.com/Layr-Labs/eigenda-proxy/pull/95

Reply:
Do you think its worth trying to track these bytes as labels then? Or should we just separate request metrics by put vs get?
Also thinking every request to a given eigenda-proxy request prob uses the same commitmentHeader?

Reply:
we should probably do (commitment mode, cert version)

Reply:
Added this as comment in [7b5aab1](https://github.com/Layr-Labs/eigenda-proxy/pull/94/commits/7b5aab10baa50adb01c7bc49f190c1adf9a55d7e)
Resolving to merge for now.